### PR TITLE
Feature/check otp

### PR DESCRIPTION
**Pull request by kkyrychenko**

### CODE REVIEW

File: src/main/java/com/icthh/xm/ms/otp/service/OneTimePasswordService.java [hunk 1]
```diff
@@ -18,6 +19,14 @@
      */
     OneTimePasswordDTO generate(OneTimePasswordDTO oneTimePasswordDTO);
 
+
+    /**
+     * Validates one time password.
+     *
+     * @param oneTimePasswordDTO the entity to save
+     */
+    void check(OneTimePasswordCheckDTO oneTimePasswordDTO);
```
Comment (ishkurko) [inline src/main/java/com/icthh/xm/ms/otp/service/OneTimePasswordService.java]: Please, use the style for abbreviations like `Dto`

File: src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java [hunk 2]
```diff
@@ -118,6 +120,27 @@ private OneTimePassword getOneTimePassword(OneTimePasswordDTO oneTimePasswordDTO
         return oneTimePassword;
     }
 
+    @Override
+    public void check(OneTimePasswordCheckDTO oneTimePasswordDTO) {
+
+        OneTimePassword passwordEntity = oneTimePasswordRepository.getOne(oneTimePasswordDTO.getId());
+
+        if (passwordEntity.getStateKey() != StateKey.ACTIVE
```
Comment (ishkurko) [inline src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java]: Could we find a more elegant way to write this condition?

Comment (vkirichenko) [reply src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java]: dome

File: src/main/java/com/icthh/xm/ms/otp/service/OneTimePasswordService.java (hunk 1 above)
Comment (ishkurko) [reply src/main/java/com/icthh/xm/ms/otp/service/OneTimePasswordService.java]: Where?

File: src/test/java/com/icthh/xm/ms/otp/web/rest/OneTimePasswordResourceIntTest.java [hunk 3]
```diff
@@ -177,18 +185,177 @@ public void testOtp() throws Exception {
...
+
+        OneTimePassword newOtp = oneTimePasswordRepository.getOne(otp.getId());
+        assertSame(StateKey.ACTIVE, newOtp.getStateKey());
+        assertEquals(newOtp.getRetries().intValue(), 4);
+
+        log.info(result.getResponse().getContentAsString());
     }
 
-    private String toJson(OneTimePasswordDTO tdo) throws JsonProcessingException {
+    private String toJson(Object tdo) throws JsonProcessingException {
```
Comment (ishkurko) [inline src/test/java/com/icthh/xm/ms/otp/web/rest/OneTimePasswordResourceIntTest.java]: tdo -> dto

File: src/test/java/com/icthh/xm/ms/otp/web/rest/OneTimePasswordResourceIntTest.java [hunk 4]
```diff
@@ -160,7 +168,7 @@ private OneTimePasswordServiceImpl getOneTimePasswordService() {
     @Test
     public void testOtp() throws Exception {
 
-        OneTimePasswordDTO tdo = new OneTimePasswordDTO();
+        OneTimePasswordDto tdo = new OneTimePasswordDto();
```
Comment (ishkurko) [inline src/test/java/com/icthh/xm/ms/otp/web/rest/OneTimePasswordResourceIntTest.java]: dto - dto

**Pull request merged**